### PR TITLE
Add base structure for Process Recipe Github Action

### PR DESCRIPTION
**Pull request by ciaransweet**
## What I'm Changing

TL;DR: This PR adds the initial skeleton of the Process Recipe GitHub Action

Current the planned flow for this Action is:

* GitHub Actions Workflow invoked when a new Recipe PR is raised
* The Workflow identifies both the `recipe.py` and `meta.yaml` files paths (**Making the assumption that it is a 1:1 relationship between PR and Recipe for now**)
* The Action takes these paths and:
    * Converts the Recipe to a Prefect Flow
    * Registers the Flow with the Bakery defined in `meta.yaml`
    * Runs a test invocation of the Flow, on the Bakery, with a subset of the input data

## How I did it

* Add `.github/actions/process_recipe`
    * `Dockerfile` to declare _what_ the Action contains (The repository is automatically mounted to `/github/workspace` by GitHub)
    * `entrypoint.sh` to wrap both installing dependencies (if required) and invoking the process_recipe logic
    * `action.yaml` used by GitHub to provide information about the Action
    * `action/process_recipe.py` - Noddy Python file which just prints the parameters passed to it
* Add `.github/workflows/process-recipe.yaml`
    * Checks out the PR code
    * Retrieves all files changed in the PR
    * Locates the `recipe.py` introduced (Errors if not found)
    * Locates the `meta.yaml` introduced (Errors if not found)
    * Calls the `process_recipe` Action which is local to the repository with the paths of the above files

## How you can test it

You can build and check out the Actions container manually yourself:

```bash
# Assuming you're in the repo root
$ cd .github/actions/process_recipe
$ docker build . -t process_recipe
$ docker run process_recipe:latest "a/fake/path.py" "a/fake/path.yaml"
recipe.py: a/fake/path.py meta.yaml: a/fake/path.yaml
```

Once this PR is merged, you can then raise a PR to test out the workflow.

To pass the `process-recipe.yaml` workflow, your PR **must** contain a `recipe.py` and `meta.yaml` file, they must both be under `recipes/` but can be however many levels of directories down that you wish inside.

For example, the below paths are perfectly **OK** in the eyes of the `process-recipe.yaml` workflow:

```
recipes/my/super/cool/recipe/recipe.py
recipes/my/super/cool/recipe/meta.yaml
```

### CODE REVIEW

**Comment by ciaransweet** [comment]
@rabernat, @sharkinsspatial mentioned that this fork & branch were the most 'current' state of affairs for `staged-recipes`. Hence the PR to your branch here.

This is the skeleton of the Action (and maybe the Workflow) that will take a recipe and convert and run it. Mainly adding so that @sharkinsspatial can add his code into it to try out

**Comment by rabernat** [comment]
Great, thanks so much for this! I don't understand a lot of what's happening, but I'll merge this here just so we have it on record.

Going forward, we will want to completely replace what's currently in the master branch of `pangeo-forge/staged-recipes` with this skeleton. So feel free to open another PR upstream, or just push directly. The master branch is not currently used for anything important, other than a base for future PRs.
